### PR TITLE
feat: add the following classes needed for channel implementation : Channel, User, UserManager

### PR DESCRIPTION
**Pull request by ppiques**
The classes are not meant to be used as is but to be a base for further implementation.

### CODE REVIEW

File: include/UserManager.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+	public:
+		typedef std::map<int, User>::iterator	iterator;
+		typedef std::map<int, User>::const_iterator	const_iterator;
+
+		UserManager(void);
+		UserManager(const UserManager &obj);
+		UserManager	&operator=(const UserManager &rhs);
+		~UserManager(void);
+
```
Comment (theovgl) [inline include/UserManager.hpp:17]: Maybe you can add some accessors like `begin()` or `end()` :)

Comment (theovgl) [reply include/UserManager.hpp:17]: And overload the operator []

File: include/User.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#include <string>
+
+class User
+{
+	public:
+		User(const std::string &name);
+		User(const User &obj);
+		User &operator=(const User &rhs);
+		~User(void);
+
```
Comment (theovgl) [inline include/User.hpp:13]: You can add the `getName()` method, it may be useful later

**Review by theovgl — CHANGES_REQUESTED** [review]

**Review by theovgl — APPROVED** [review]

File: include/User.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#ifndef USER_HPP
+# define USER_HPP
+
+#include <string>
```
Comment (vfurmane) [inline include/User.hpp:4]: image

File: include/UserManager.hpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#ifndef USERMANAGER_HPP
+# define USERMANAGER_HPP
+
+#include <map>
+#include "User.hpp"
```
Comment (vfurmane) [inline include/UserManager.hpp:5]: image

File: include/User.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+	public:
+		User(const std::string &name);
+		User(const User &obj);
+		User &operator=(const User &rhs);
+		~User(void);
+
+		const std::string	&getName(void) const;
+	
+	private:
+		std::string	_name;
```
Comment (vfurmane) [inline include/User.hpp:17]: The `User` class should have a reference to a Peer and the name would be stored in it. It prevents duplication of code.

**Review by vfurmane — CHANGES_REQUESTED** [review]

File: include/Channel.hpp [hunk 6]
```diff
@@ -1,8 +1,10 @@
 #ifndef CHANNEL_HPP
 # define CHANNEL_HPP
 
-#include <string>
-#include "UserManager.hpp"
+# include <stdint.h>
+# include <string>
+# include <stdint.h>
```
Comment (vfurmane) [inline include/Channel.hpp:6]: Plouto do foi koune

**Review by vfurmane — CHANGES_REQUESTED** [review]

**Review by vfurmane — APPROVED** [review]
Could be better....

**Review by theovgl — APPROVED** [review]

**Pull request merged**